### ISSUE TITLE
Use oAuth 2 endpoints for all APIs

### ISSUE THREAD

**Issue by kimar** [Geofancy/ios-app#28]
Use this (or similar) for Authentication:

https://github.com/raulriera/AuthenticationViewController

**Comment by francisdb** [comment]
Not sure this will give good UX. OAuth2 also has a  grant_type=password flow but would be best to only allow that for the app.

**Comment by kimar** [comment]
I'm not sure if the password flow has been implemented in our oAuth strategy yet. I'd also rather take this as a long term goal to unify (allthethings) instead of rushing on that topic. But I'd really like to see every app as an api consumer instead of maintainig special private apis.

**Comment by francisdb** [comment]
I agree, but showing a web page for auth is not always ideal as it feels different compared to a native form. Anyway, just mention that there is an option to keep password/email login even with OAuth2

**Comment by kimar** [comment]
Thanks, I'll definitely have a look into this. The idea of that web form is that you're seeing where you're entering your credentials. But as it's the official app this is a minir I'd reckon. 👍 let's see how we proceed on this. Maybe this is a good one for A/B testing 😂

**Comment by francisdb** [comment]
passport surely should have a plugin for handling grant_type=password oauth2

**Comment by francisdb** [comment]
Anyway this should be moved to the server repo ticketing

**Comment by kimar** [comment]
There's already a ticket on the server side for this, this ticket was just
my reminder for the intergeational side of things in the app itself.
